### DISCUSSION TITLE
Christian's Templates

### DISCUSSION THREAD

**Discussion by chhoumann** [SilentVoid13/Templater#230] [Templates Showcase]
Hey everyone. I'm sharing my templates here:
https://github.com/chhoumann/Templater_Templates

You can see pictures and examples of them in action. They should be very customizable.
**Templates**
- Readwise importer - Import anything from your Readwise vault. Anything, anytime. Also, you can import just one highlight at a time.
- QuickAdd - Add new pages from anywhere, based on a specified template. You can define a start symbol, so if your litterature notes are written like `{ Book Note`, you can specify that - and you'll only have to enter the note name, not `{`.
- Latex templates for creating new environments and adding colors. These are selected using Suggesters, so you won't have 10 templates for each color.
- Code snippet template - Add a new snippet through a suggester with predefined languages

**Comment by JeppeKlitgaard** [comment]
Thank you @chhoumann 

I just replaced my LaTeX env setup with yours based suggesters. I hadn't thought to use those!

I would suggest adding `<% tp.file.selection() %>` before the cursor to (optionally) wrap around the selected content. This combined with a hotkey makes for a very ergonomic LaTeX setup.

Full modified version:
```
<%* 
const envs = [
	"gather",
	"equation",
	"align",
	"matrix",
	"array",
	"cases",
	"tabular",
];
const envType = await tp.system.suggester(envs,envs); 
if (envType) {
_%>

\begin{<%envType%>}
	<% tp.file.selection() %><% tp.file.cursor(1) %>
\end{<% envType %>}<%*} _%>
```

**Reply by chhoumann** [reply]
Brilliant! That's great. Thank you. I've updated the template 😄

**Comment by chhoumann** [comment]
QuickAdd has received a v2 update. Check it out here:
https://github.com/chhoumann/Templater_Templates#quickadd-v2

Quickly add new pages or content to your vault. Powerful and customizable Templater 'plugin'.
This template allows you to quickly add new pages from templates, as well as quick capture anything. You can customize practically everything about it. There's even custom templates, just for formatting titles and values.
